src/gui/sdlmain.cpp

```diff
@@ -3827,17 +3817,16 @@ bool GFX_Events()
 		return true;
 	last_check = current_check;
 #endif
-
-	SDL_Event event;
-#if defined (REDUCE_JOYSTICK_POLLING)
 	static auto last_check_joystick = GetTicks();
-	auto current_check_joystick = GetTicks();
-	if (GetTicksDiff(current_check_joystick, last_check_joystick) > 20) {
-		last_check_joystick = current_check_joystick;
-		if (MAPPER_IsUsingJoysticks()) SDL_JoystickUpdate();
-		MAPPER_UpdateJoysticks();
+	if (MAPPER_IsUsingJoysticks()) {
+		auto current_check_joystick = GetTicks();
+		if (GetTicksDiff(current_check_joystick, last_check_joystick) > 20) {
+			last_check_joystick = current_check_joystick;
+			SDL_JoystickUpdate();
+			MAPPER_UpdateJoysticks();
+		}
```

Comment:
Commenting on your latest changes now.  You've reverted most of your changes to this function but again we have a divergence from `main` with no explanation as to why.

Previous code always ran `MAPPER_UpdateJoysticks()` when the duration exceeded 20ms.  You've now put this entire block behind `MAPPER_IsUsingJoysticks()`.  It makes logical sense that we shouldn't be updating joysticks if we're not using any but the mapper is a pile of spaghetti and this could very well break something.

Again, I'm just going to insist that any significant changes to this function be in a separate commit with an explanation.

Reply:
Reverted the rest except the `#if defined REDUCE_JOYSTICK_POLLING`